src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderList.jsx

```diff
@@ -109,144 +113,143 @@ export default function ProviderList({
           sortByDistanceFromResidential={sortByDistanceFromResidential}
         />
       )}
-      {loadingProviders && (
-        <div className="vads-u-padding-bottom--2">
-          <LoadingIndicator message="Loading the list of providers." />
-        </div>
-      )}
+      {loadingProviders &&
```

Comment:
This change is the main fix to remove the extra loading provider spinner during dual loading states. The rest is refactoring the conditional render logic.